src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```
@@ -140,13 +140,9 @@ public void clusterDescriptionChanged(
           if (!latch.await(latchTimeout, TimeUnit.MILLISECONDS)) {
             configValue.addErrorMessage("Unable to connect to the server.");
             mongoClient.close();
```

Do we want such condition for Non srv protocol host (self-managed instance) as well?

---

This validation already happens in fail fast validation

---

AFAIR -> The fail fast triggers the connector validation itself during the flow.

---

Could you try out a scenario with an invalid self-managed instance?

---

sorry, I mean't connect templates
Connect templates suffices for this but source connector does not go through that flow as it does not have region validation. Hence adding it here makes sense.